packages/react-query/src/types.ts

```diff
@@ -28,19 +29,19 @@ export interface UseBaseQueryOptions<
   TQueryData = TQueryFnData,
   TQueryKey extends QueryKey = QueryKey,
 > extends ContextOptions,
-    QueryObserverOptions<TQueryFnData, TError, TData, TQueryData, TQueryKey> {}
+    WithRequired<
+      QueryObserverOptions<TQueryFnData, TError, TData, TQueryData, TQueryKey>,
+      'queryKey'
+    > {}
```

Comment:
why don't we make the `queryKey` required directly inside `QueryOptions` ?

https://github.com/TanStack/query/blob/2b01ac0828b8dcabfa1a1747f44aaeb11e9c28ad/packages/query-core/src/types.ts#L76

Is there ever a situation where we want the `queryKey` to be optional there? I think it was only optional because we had overloads and the key could be a separate parameter ...

Reply:
`QueryOptions` is used everywhere and extended by many things .. As a safe option, I made the `queryKey` only required in the places we need it.

Here are some examples where I think we don't need the queryKey to be required:

![default options](https://user-images.githubusercontent.com/36894846/209786489-a913f9d9-abfe-4650-926e-9da9026fa016.png)

![HydrateOptions](https://user-images.githubusercontent.com/36894846/209786963-f1b9eb16-b32b-4b38-a5d7-fe7c0e36c379.png)

![getNextPageParam](https://user-images.githubusercontent.com/36894846/209787085-28cb0ebf-c677-43ca-ade0-4f05ecbb9e7f.png)

![fetch-In query-core/query.ts](https://user-images.githubusercontent.com/36894846/209787942-fce171fa-0d23-4beb-9de6-aaf98678cbbb.png)

![getDefaultState in query-core/query.ts](https://user-images.githubusercontent.com/36894846/209788035-72032ce5-bb00-468c-b09e-f5ea0d98c08d.png)

Reply:
that's fine, we can revisit this later if necessary. There are some functions that don't really need the full `QueryOptions` (like `getNextPageParam`).